wallet/src/de/schildbach/wallet/ui/MoreFragment.kt

```diff
@@ -66,14 +78,54 @@ class MoreFragment : Fragment(R.layout.activity_more) {
                     WalletApplication.getInstance()).show()
         }
 
+        update_profile_status_container.visibility = View.GONE
+
         val blockchainIdentity = PlatformRepo.getInstance().getBlockchainIdentity()
         if (blockchainIdentity != null && blockchainIdentity.registrationStatus == BlockchainIdentity.RegistrationStatus.REGISTERED) {
-            AppDatabase.getAppDatabase().dashPayProfileDaoAsync().loadByUserIdDistinct(blockchainIdentity.uniqueIdString)
+            editProfileViewModel.dashPayProfileData
                     .observe(viewLifecycleOwner, Observer {
                         if (it != null) {
+                            dashPayProfile = it
                             showProfileSection(it)
                         }
                     })
+
+            //show updating profile animation if necessary
+
+            editProfileViewModel.updateProfileRequestState.observe(context!!.lifecycleOwner()!!, Observer {
+                if (it != null && it.containsKey(dashPayProfile.userId)) {
+                    val state = it[dashPayProfile.userId]
+                    when(state!!.status) {
+                        Status.SUCCESS -> {
+                            Toast.makeText(requireActivity(), "Update successful", Toast.LENGTH_LONG).show()
+                            update_profile_status_container.visibility = View.GONE
+                            editProfile.visibility = View.VISIBLE
+                        }
+                        Status.ERROR -> {
+                            var msg = state.message
+                            if (msg == null) {
+                                msg = "!!Error!!  ${state.exception!!.message}"
+                            }
+                            Toast.makeText(requireActivity(), msg, Toast.LENGTH_LONG).show()
+                            update_profile_status_container.visibility = View.VISIBLE
+                            update_status_text.text = msg
+                            editProfile.visibility = View.VISIBLE
+                        }
+                        Status.LOADING -> {
+                            Toast.makeText(requireActivity(), "Processing update", Toast.LENGTH_LONG).show()
+                            update_profile_status_container.visibility = View.VISIBLE
+                            editProfile.visibility = View.GONE
+                        }
+                        Status.CANCELED -> {
+                            update_profile_status_container.visibility = View.VISIBLE
+                            update_status_text.text = "Cancelled" //hard coded text
+                            editProfile.visibility = View.VISIBLE
+                        }
+                    }
```

Comment:
This observer seems to be called after a update operation is finished in the EditProfileActivity and the user returns to the More Screen.